### PR TITLE
WL-4264, WL-4265, WL-4249: Export link, Print link, Last updated text

### PR DESCRIPTION
**Pull request by nicholaswilson100**

### CODE REVIEW

File: citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java [hunk 1]
```diff
@@ -340,13 +335,28 @@ protected void handleViewRequest(HttpServletRequest req, HttpServletResponse res
 
     		List<Citation> citations = collection.getCitations();
     		String contentCollectionId = resource.getContainingCollection().getId();
-    		String exportUrlAll = collection.getUrl(org.sakaiproject.citation.api.CitationService.REF_TYPE_EXPORT_RIS_ALL) + "?citationCollectionId=" + citationCollectionId + "&resourceDisplayName=" + title + "&contentCollectionId=" + contentCollectionId;
+    		String exportParams =  "?resourceDisplayName=" + resource.getProperties().getProperty(ResourceProperties.PROP_DISPLAY_NAME) + "&resourceId=" + resource.getId();
+    		String exportUrlAll = fullCollection.getUrl(org.sakaiproject.citation.api.CitationService.REF_TYPE_EXPORT_RIS_ALL) + exportParams;
+    		String displayDate = null;
+    		try {
+    			displayDate = new SimpleDateFormat("dd/MM/yyyy").format(properties.getDateProperty(ResourceProperties.PROP_MODIFIED_DATE));
```
Comment (buckett) [inline citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java]: i18n?

File: citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java [hunk 2]
```diff
@@ -340,13 +335,28 @@ protected void handleViewRequest(HttpServletRequest req, HttpServletResponse res
...
+    		} catch (EntityPropertyNotDefinedException e) {
+    			m_log.warn("CitationListAccessServlet.handleViewRequest() : Property name requested is not defined - for contentCollectionId" + contentCollectionId);
```
Comment (buckett) [inline citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java:345]: Should these be warn in production?

**Comment by nicholaswilson100** [comment]
This request contains 3 JIRAs with closely related code (hence same request):

WL-4264: Adding Export link to top right of student view of reading list.
WL-4254: Adding Print link to top right of student view of reading list (which is meant to open a new tab with the same reading list but everything expanded).
WL-4249: Adding 'Last updated' date  to top right of student view of reading list.

File: citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java (hunk 2 above)
Comment (nicholaswilson100) [inline citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java:345]: It seems more important than info but less important than error - I don't mind changing it to different level.

Comment (buckett) [inline citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java:345]: Is it the case that they will always have a modification date?

Comment (nicholaswilson100) [inline citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java:345]: Yes last modified date is the creation date when a list is first created

Comment (nicholaswilson100) [inline citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java:345]: unless something goes wrong of course

**Comment by nicholaswilson100** [comment]
Linked to https://github.com/ox-it/wl-reference/pull/124

File: citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java (hunk 2 above)
Comment (buckett) [inline citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java:345]: Ok, your right, `warn` is the way to go.

File: citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java [hunk 3]
```diff
@@ -340,13 +335,28 @@ protected void handleViewRequest(HttpServletRequest req, HttpServletResponse res
 
     		List<Citation> citations = collection.getCitations();
     		String contentCollectionId = resource.getContainingCollection().getId();
-    		String exportUrlAll = collection.getUrl(org.sakaiproject.citation.api.CitationService.REF_TYPE_EXPORT_RIS_ALL) + "?citationCollectionId=" + citationCollectionId + "&resourceDisplayName=" + title + "&contentCollectionId=" + contentCollectionId;
+    		String exportParams =  "?resourceDisplayName=" + resource.getProperties().getProperty(ResourceProperties.PROP_DISPLAY_NAME) + "&resourceId=" + resource.getId();
+    		String exportUrlAll = fullCollection.getUrl(org.sakaiproject.citation.api.CitationService.REF_TYPE_EXPORT_RIS_ALL) + exportParams;
+    		String displayDate = null;
+    		try {
+    			displayDate = new SimpleDateFormat("dd/MM/yyyy", rb.getLocale()).format(properties.getDateProperty(ResourceProperties.PROP_MODIFIED_DATE));
```
Comment (buckett) [inline citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java]: I'm not sure that passing a locale will do anything here as the issue is that different countries format their dates differently. You should let the locale pick the date format.
https://confluence.sakaiproject.org/display/I18N/Best+Practices+for+Internationalized+Tools+in+Sakai

**Comment by buckett** [comment]
If you push a new commit I don't get notified so don't see the changes it's only when you make a comment I'll come back to it.

**Comment by nicholaswilson100** [comment]
Yeah I know. I wasn't really happy with the change so I was looking for something better which is there now - thanks for the confluence link!

**Pull request merged**